### PR TITLE
Update release_chart.yaml to run on dev branch

### PR DESCRIPTION
**Pull request by jklugherz**

### CODE REVIEW

File: .github/workflows/release_dev_chart.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          helm repo add bitnami https://charts.bitnami.com/bitnami
+
+      - name: Run chart-releaser
+        uses: helm/chart-releaser-action@v1.3.0
+        with:
+          mark_as_latest: false
+        env:
+          CR_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          CR_SKIP_EXISTING: "true"
+          CR_RELEASE_NAME_TEMPLATE: "{{ .Name }}-{{ .Version }}-dev"
```
Comment (bpblanken) [inline .github/workflows/release_dev_chart.yaml:41]: is this the only line of difference?  if so we can probably template out the "-dev".

File: .github/workflows/release_dev_chart.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.9.2
+
+      - name: Seed helm deps
+        run: |
+          helm repo add bitnami https://charts.bitnami.com/bitnami
+
+      - name: Run chart-releaser
```
Comment (jklugherz) [inline .github/workflows/release_dev_chart.yaml:34]: The important stuff:
> mark_as_latest: false

don't mark as latest release in github

> CR_RELEASE_NAME_TEMPLATE
add -dev suffix to the release name

File: .github/workflows/release_dev_chart.yaml (hunk 1 above)
Comment (jklugherz) [reply .github/workflows/release_dev_chart.yaml:41]: this line and `mark_as_latest: false` are the only differences

Comment (bpblanken) [reply .github/workflows/release_dev_chart.yaml:41]: I did something that might be relevant [here](https://github.com/broadinstitute/seqr/blob/master/.github/workflows/hail-search-persistent-volume-snapshot-release.yaml#L65)

File: charts/seqr/templates/deployment.yaml [hunk 3]
```diff
@@ -54,12 +54,14 @@ spec:
...
         {{- include "seqr.environment" $ | nindent 8 }}
         livenessProbe:
+          initialDelaySeconds: {{ .Values.probeInitialDelaySeconds }}
           exec:
             command:
             - /bin/bash
             - -c
             - /readiness_probe
         readinessProbe:
+          initialDelaySeconds: {{ .Values.probeInitialDelaySeconds }}
```
Comment (bpblanken) [inline charts/seqr/templates/deployment.yaml:64]: I think pulling `main` into `dev` will remove this diff!

File: .github/workflows/release_chart.yaml [hunk 4]
```diff
@@ -31,8 +32,21 @@ jobs:
...
+            echo "CR_RELEASE_NAME_TEMPLATE={{ .Name }}-{{ .Version }}-dev" >> $GITHUB_ENV
+          else
+            echo "MARK_AS_LATEST=true" >> $GITHUB_ENV
+            echo "CR_RELEASE_NAME_TEMPLATE={{ .Name }}-{{ .Version }}" >> $GITHUB_ENV
+          fi
+
       - name: Run chart-releaser
         uses: helm/chart-releaser-action@v1.3.0
+        with:
+          mark_as_latest: ${{ env.MARK_AS_LATEST }}
```
Comment (bpblanken) [inline .github/workflows/release_chart.yaml:48]: what happens if you just do:

```
mark_as_latest: "${{ github.event.workflow_run.head_branch }}" == "main"
```

Comment (jklugherz) [reply .github/workflows/release_chart.yaml:48]: that should work, is there any way to test these before they're merged in?

Comment (jklugherz) [reply .github/workflows/release_chart.yaml]: I found and ran `act` with your suggestion and got an error: ```Error: workflow is not valid. 'release_chart.yaml': yaml: line 47: did not find expected key```!

Comment (jklugherz) [reply .github/workflows/release_chart.yaml:48]: here's the `act` output for the step I added:

[Release Charts/release]   ✅  Success - Main Set environment variables for chart-releaser
[Release Charts/release]   ⚙  ::set-env:: MARK_AS_LATEST=true
[Release Charts/release]   ⚙  ::set-env:: CR_RELEASE_NAME_TEMPLATE={{ .Name }}-{{ .Version }}

Comment (bpblanken) [reply .github/workflows/release_chart.yaml:48]: what if we the syntax from my suggestion to something like:

```
"${{ github.event.workflow_run.head_branch  == 'main' }}"
```

using `MARK_AS_LATEST` seems more verbose than required, at least to my eyes!

Comment (bpblanken) [reply .github/workflows/release_chart.yaml:48]: Good find on `act`! I've seen it floating around but never tried it.  historically testing GitHub actions has been the actual worst so good to know that there's an option that seems to work!

**Review by bpblanken — APPROVED** [review]
Nit, but could you also one-liner the template?

```
CR_RELEASE_NAME_TEMPLATE: "{{ .Name }}-{{ .Version }}${{ github.event.workflow_run.head_branch == 'dev' && '-dev' || '' }}
```

**Pull request merged**